source/js/RecipeCard.js

```diff
@@ -0,0 +1,251 @@
+class RecipeCard extends HTMLElement {
+  constructor() {
+    super();
+    let shadow = this.attachShadow({mode: 'open'});
+  }
+
+  set data(data) {
+    const style=`
+    .recipe-card {
+      width: 300px;
+
+      display: grid;
+      grid-template-rows: [top] 50% [image-bottom] 1.5em [title-bottom] 1.5em [info-bottom]  [bottom];
+      grid-template-columns: [left] auto [right];
+
+      background: #FFF6EC;
+    }
+
+    .thumbnail-photo {
+      height: 225px;
+      object-fit: cover;
+      width: 100%;
+    }
+
+    .rating-time {
+      display: grid;
+      grid-template-rows: [top] auto [bottom];
+      grid-template-columns: [left] 50% [middle] 50% [right];
+    }
+    `;
+    const styleElem = document.createElement('style');
+    styleElem.innerHTML=style;
+
+
+    const cleanData = {};
+    // process raw data into cleanData
+    cleanData.thumbnail = searchForKey(data,'thumbnailUrl');
+    cleanData.title = getRecipeTitle(data);
+    cleanData.url = getUrl(data);
+    cleanData.organization = getOrganization(data);
+    let cookTime = searchForKey(data,'cookTime');
+    let prepTime = searchForKey(data,'prepTime');
+    let totalTime = searchForKey(data,'totalTime');
+    cleanData.cookTime = convertTime(cookTime);
+    cleanData.prepTime = convertTime(prepTime);
+    cleanData.totalTime = convertTime(totalTime);
+    cleanData.ingredients = createIngredientList(searchForKey(data,'recipeIngredient'));
+    let tempRating = searchForKey(data,'aggregateRating');
+    if (!!tempRating) {
+      cleanData.rating = {
+        count: tempRating.ratingCount,
+        score: tempRating.ratingValue
+      }
+    }
+
+    const card = document.createElement('article');
+    card.classList.add('recipe-card');
+
+    const photo = document.createElement('img');
+    photo.classList.add("thumbnail-photo");
+    photo.setAttribute('src', cleanData.thumbnail);
+    card.appendChild(photo);
+
+    const title = document.createElement('p');
+    title.textContent = cleanData.title;
+    card.appendChild(title);
+
+    const info = document.createElement('div');
+    info.classList.add('rating-time');
+
+    const rating = document.createElement('p');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
    const rating = document.createElement("p");
```